### PR TITLE
Add pytorch support

### PR DESCRIPTION
**Pull request by georgepar**
Hi all, 

This PR adds some bindings for PyTorch:

- optimizers
- activation functions
- loss functions
- learning schedules
- distributions
- data types

### CODE REVIEW

**Comment by sguada** [comment]
Thanks for your contribution, we will take a closer look and get back to you.

**Comment by sguada** [comment]
Can you rebase this PR with the latests changes?

**Comment by valentindey** [comment]
Hi! I'm really interested in this feature. I see that the pull request is pending for quite some time now. Are there any plans on how to proceed? I'm also happy to jump in.

**Comment by valentindey** [comment]
I created an updated version of this (#35 ) since I am not able to change this one. Thanks a lot to @georgepar for his great work! I hope @sguada or @dhr can give it a look sometime soon :slightly_smiling_face:

**Comment by sguada** [comment]
If you are interested in submitting this PR, please let us know.

**Comment by dhr** [comment]
Sorry for the long delay on this. We've added some automated GitHub syncing tools which will improve our ability to accept PRs like this one going forward. We've merged the updated version of this PR in #35, so closing this PR now. Thanks @georgepar for your work on this!
